app/shop.py

```diff
@@ -0,0 +1,32 @@
+import datetime
+
+from typing import Any
+
+
+class Shop:
+    def __init__(
+            self,
+            name: str,
+            location: list,
+            products: dict
+    ) -> None:
+        self.name = name
+        self.location = location
+        self.products = products
+
+    def print_receipt(self, customer: Any) -> None:
+        timestamp = datetime.datetime.now().strftime("%d/%m/%Y %H:%M:%S")
+        print(f"Date: {timestamp}")
+        print(f"Thanks, {customer.name}, for your purchase!")
+        print("You have bought:")
+        total_cost = 0
+        for product, quantity in customer.product_cart.items():
+            cost = self.products[product] * quantity

```

Comment:
Ensure that the product exists in `self.products` before attempting to calculate the cost. You might want to add a check to handle cases where a product in the customer's cart is not available in the shop's inventory.